### DISCUSSION TITLE
Help With MacOS Install

### DISCUSSION THREAD

**Discussion by ZITNALTA** [kreneskyp/ix#51] [Q&A] [answered]
Hi, I just finished watching a YT video "iX : GPT-4 Autonomous Ai Agent Platform" and I got 75%+ of the way to completion but apparently I did not have Docker Installed on my MacBook Pro (M1). I did get all the API Keys setup in VSCode .env. 

Can someone direct me to a MacOS tutorial on how to set this up? Thanks.

**Comment by kreneskyp** [comment]
I don't have a mac to test on but I'll help how I can.  

What step did you get stuck at? Any error messages?

**Reply by ZITNALTA** [reply]
Thanks for your reply! I went thru the video from WorldOfAI to the 10:53 mark and then I tried the "pip install -r requirements.txt" command but it did not work. I believe it is already installed (not sure) because I installed python. I recall it saying there was a newer version of pip, but I did not install it. I did get this error "command not found: pip" when I tried that command in the "ix" directory in the MacOS Terminal. At the 13:40 of the video mark I tried installing the dev image and that is where I realized I did not have Docker installed. So I installed Docker.dmg. This is where I got stuck. So to recap, 1) I am not sure I did the pip thing correctly and 2) I installed Docker manually from the Docker.dmg download. I also tried to install it through VSCode and then tried to install it through the Terminal from instructions on the Docker website.

**Reply by kreneskyp** [reply]
I hadn't seen that video so I'm not sure it's providing the correct install instructions.

1. First setup Docker desktop for mac:
    https://docs.docker.com/desktop/install/mac-install/
    
2. Follow the install instructions in the readme.
    https://github.com/kreneskyp/ix#build-and-run-the-dev-image

**Reply by ZITNALTA** [reply]
I successfully installed Docker desktop based on those instructions so I am all set there. Here is the error message I get when I run the make dev_setup command in the Terminal on my MacBook: 
/bin/sh: md5sum: command not found
/bin/sh: md5sum: command not found
/bin/sh: md5sum: command not found
echo building ghcr.io/ix/sandbox:
building ghcr.io/ix/sandbox:
docker build -t ghcr.io/ix/sandbox: -f Dockerfile .
[+] Building 0.0s (0/0)                                                         
ERROR: invalid tag "ghcr.io/ix/sandbox:": invalid reference format
make: *** [.sentinel/image] Error 1

As far as the video that is why I asked if there was a video on how to install on MacOS because that video is for windows.

**Reply by kreneskyp** [reply, marked as answer]
> /bin/sh: md5sum: command not found

I just merged a fix for this bug.   If you `git pull origin master` it should pull in the fix.

**Comment by ZITNALTA** [comment]
![IX localhost - Screenshot 2023-05-24 at 10-14-12 Ix](https://github.com/kreneskyp/ix/assets/127397646/2750ba26-7800-4d1b-9c71-278a09c79f94)

**Reply by ZITNALTA** [reply]
Worked, thanks!

**Reply by kreneskyp** [reply]
Fantastic!

**Reply by ZITNALTA** [reply]
I spoke too soon. The interface is all setup but I can't get it to execute any of my chat requests. I did go in and configure it so that GPT-4 was selected for the three agents (code, planner, ix). Not getting any error message just not getting anything after chat prompt is submitted. Is there a max length of characters for the chat submission prompt or something? Sorry to bother you again with this.
